draft-bormann-asdf-instance-information.md

```diff
@@ -496,25 +469,38 @@ namespace:
 defaultNamespace: cap
 sdfObject:
   temperatureSensor:
+    sdfContext:
+      ipAddress:
+        type: string
+        format: TODO
     sdfProperty:
       temperature:
         description: Temperature value measure by this Thing's temperature sensor.
         type: number
         unit:
-          "$comment": Should schema information be settable via a constructor at all? This question might indicate that we need different kinds of constructors
           type: string
     sdfAction:
       construct:
         sdfInputData:
-          "$comment": "DISCUSS: How can we establish a connection between constructor parameters and target properties?"
+          sdfRelation:
+            constructor:
+              "$comment": Do we need to import the relation type from a namespace definition?
+              relType: constructor
+              target: '#/sdfObject/temperatureSensor'
           type: object
           properties:
             temperatureUnit:
               type: string
-              target: "#/sdfObject/temperatureSensor/sdfProperty/temperature/unit"
+              sdfRelation:
+                parameter:
+                  relType: constructorParameter
+                  target: '#/sdfObject/temperatureSensor/sdfProperty/temperature/unit'
             ipAddress:
-              "$comment": How can we express that this is context information?
-              isContextInformation: true
+              type: string
+              sdfRelation:
+                paramter:
+                  relType: constructorParameter
+                  target: '#/sdfObject/temperatureSensor/sdfContext/ipAddress'
           required:
             - temperatureUnit
         sdfOutputData:
```

Comment:
Regarding constructors, I had two additional ideas in the meantime:

1. We could simply model different constructors by offering different sets of `sdfInputData` parameters via `sdfChoice`. This assumes, though, that the output format would stay consistent (e.g., since we are only returning a pointer to the result).
2. The preference of constructors could also be modeled by using `sdfRelation`, although the details need to be discussed here – should there be transitive `sdfRelations` between constructors, with `sdfRelation`s between the “adjacent” ones? Or could we also have an `sdfRelation` with multiple `target`s and then apply a mapping of priorities to that?